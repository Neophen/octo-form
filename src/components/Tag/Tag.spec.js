import { shallowMount } from "@vue/test-utils";
import OTag from "./Tag";

let wrapper;

describe("OTag", () => {
  beforeEach(() => {
    wrapper = shallowMount(OTag);
  });

  it("is called", () => {
    expect(wrapper.name()).toBe("OTag");
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("emit close event when closing if not disabled", () => {
    wrapper.setProps({ disabled: false });

    const close = jest.fn();
    wrapper.vm.$on("close", close);
    wrapper.vm.close();

    wrapper.setProps({ disabled: true });
    wrapper.vm.close();

    expect(close).toHaveBeenCalledTimes(1);
  });
});
