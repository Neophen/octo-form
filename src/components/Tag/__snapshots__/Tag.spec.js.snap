// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTag render correctly 1`] = `
<span class="octo-tag"><span class=""></span>
<!----></span>
`;
